### PR TITLE
Raise an error when gconstruct reads empty files.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/file_io.py [hunk 1]
```diff
@@ -132,6 +135,8 @@ def read_data_json(data_file, data_fields):
         for line in json_file.readlines():
             record = json.loads(line)
             data_records.append(record)
+    assert len(data_records) > 0, \
+        f"{data_file} is empty {data_records}."
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/file_io.py:139]: if it's empty, why do you need to include data_records in the message?

File: python/graphstorm/gconstruct/file_io.py [hunk 2]
```diff
@@ -80,6 +80,9 @@ def read_data_csv(data_file, data_fields=None, delimiter=','):
     dict of Numpy arrays.
     """
     data = pd.read_csv(data_file, delimiter=delimiter)
+    assert data.shape[0] > 0, \
+        f"{data_file} has an empty data. The data frame shape is {data.shape}"
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/file_io.py:84]: if the file is empty, what is the shape of the data?

**Review by zheng-da — APPROVED** [review]

File: python/graphstorm/gconstruct/file_io.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/file_io.py:84]: It still contains the column names.

File: python/graphstorm/gconstruct/file_io.py (hunk 1 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/file_io.py:139]: It still contains the column names.

**Pull request merged**